### PR TITLE
[@types/react-dates] Fix incorrect phrase arg type and add missing phrase properties

### LINKED ISSUE #39929: [@types/react-dates] chooseAvailableStartDate & chooseAvailableEndDate prop types incorrect
If you know how to fix the issue, make a pull request instead.

- [X] I tried using the `@types/react-dates` package and had problems.
- [X] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [X] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [X] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @ArturAmpilogov @NathanNZ

If you do not mention the authors the issue will be ignored.

`chooseAvailableStartDate?: (date: string) => string;`
`chooseAvailableEndDate?: (date: string) => string;`

I believe the prop should be an object:
https://github.com/airbnb/react-dates/blob/master/src/defaultPhrases.js#L29

### PR DESCRIPTION
**Pull request by gidztech**
Fixes: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/39929

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/airbnb/react-dates/blob/master/src/defaultPhrases.js#L29
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-dates ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-dates@latest/))
was missing the following properties:
1. CalendarDay
1. CalendarMonth
1. CalendarMonthGrid
1. DateRangePickerInput
1. DateRangePickerInputController

<details>
<summary>as well as these 4 other properties...</summary>
<p>DateRangePickerShape, DayPicker, SingleDatePickerInput, SingleDatePickerShape</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b5c2ee6fdf109bbe234510c0f85650271b784f96
</p>

File: types/react-dates/tslint.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
         "semicolon": false,
         "trim-file": false,
         "typedef-whitespace": false,
-        "whitespace": false
```
Comment (amcasey) [inline types/react-dates/tslint.json:20]: Why was the new lint exception required?

**Review by amcasey — APPROVED** [review]

File: types/react-dates/tslint.json (hunk 1 above)
Comment (gidztech) [reply types/react-dates/tslint.json:20]: It looks like a new rule was added globally which doesn't like how the namespace is exported as a whole.

**Comment by gidztech** [comment]
Ready to merge

**Pull request merged**